### PR TITLE
Scala and Python API for Describe Detail

### LINKED ISSUE #1188: [Feature Request] Scala and Python API for Describe Detail
### Overview

`DESCRIBE DETAIL` is supported in SQL. But similar to `DESCRIBE HISTORY` which has Scala and Python APIs as `DeltaTable.history()`, there should be `DeltaTable.details()`

### Willingness to contribute

The Delta Lake Community encourages new feature contributions. Would you or another member of your organization be willing to contribute an implementation of this feature?

- [ ] Yes. I can contribute this feature independently.
- [ ] Yes. I would be willing to contribute this feature with guidance from the Delta Lake community.
- [x] No. I cannot contribute this feature at this time.

### PR DESCRIPTION
**Pull request by hedibejaoui**
## Description
Resolves #1188

### PR changes 
Added a `details()` function to the Python and Scala APIs of `DeltaTable` similar to running `DESCRIBE DETAIL` to get information about a table (name, format, size, etc.) 

### Why we need the changes
Just like we have `DeltaTable.history()`, there should be a `DeltaTable.details()`

## How was this patch tested?
- Scala unit tests were added to [DescribeDeltaDetailSuite.scala](https://github.com/hedibejaoui/delta/blob/70cf2656b44798136edf39a46678fa1de2b67f0b/core/src/test/scala/org/apache/spark/sql/delta/DescribeDeltaDetailSuite.scala#L61) 
- Python unit tests were added to [tables.py](https://github.com/hedibejaoui/delta/blob/70cf2656b44798136edf39a46678fa1de2b67f0b/python/delta/tables.py#L281)
- Extended Scala and Python utilities examples and checked output of `python3 run-integration-tests.py --use-local`:
```bash
[info] Describe Details for the table
[info] +------+--------------------+----+-----------+--------------------+--------------------+--------------------+----------------+--------+-----------+----------+----------------+----------------+
[info] |format|                  id|name|description|            location|           createdAt|        lastModified|partitionColumns|numFiles|sizeInBytes|properties|minReaderVersion|minWriterVersion|
[info] +------+--------------------+----+-----------+--------------------+--------------------+--------------------+----------------+--------+-----------+----------+----------------+----------------+
[info] | delta|13a60c59-2527-4cc...|null|       null|file:/tmp/parquet...|2022-07-02 15:13:...|2022-07-02 15:13:...|              []|       6|       2656|        {}|               1|               2|
[info] +------+--------------------+----+-----------+--------------------+--------------------+--------------------+----------------+--------+-----------+----------+----------------+----------------+
..
######## Describe details for the table ######
+------+--------------------+----+-----------+--------------------+--------------------+--------------------+----------------+--------+-----------+----------+----------------+----------------+
|format|                  id|name|description|            location|           createdAt|        lastModified|partitionColumns|numFiles|sizeInBytes|properties|minReaderVersion|minWriterVersion|
+------+--------------------+----+-----------+--------------------+--------------------+--------------------+----------------+--------+-----------+----------+----------------+----------------+
| delta|46d665c3-b2bd-4de...|null|       null|file:/tmp/delta-t...|2022-07-02 15:27:...|2022-07-02 15:27:...|              []|       6|       2656|        {}|               1|               2|
+------+--------------------+----+-----------+--------------------+--------------------+--------------------+----------------+--------+-----------+----------+----------------+----------------+
``` 

## Does this PR introduce _any_ user-facing changes?
No

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/sql/delta/DescribeDeltaDetailSuite.scala [hunk 1]
```diff
@@ -58,15 +58,38 @@ trait DescribeDeltaDetailSuiteBase extends QueryTest
       Seq("format", "partitionColumns", "numFiles"))
   }
 
-  test("delta table: path") {
+  test("delta table: Scala details using table path") {
```
Comment (allisonport-db) [inline core/src/test/scala/org/apache/spark/sql/delta/DescribeDeltaDetailSuite.scala:61]: Can we instead just modify `describeDeltaDetailTest` to check both the SQL command and scala `details()`?

**Review by allisonport-db — COMMENTED** [review]
left a minor comment. thanks for updating the integration tests!

**Comment by allisonport-db** [comment]
@hedibejaoui Could you update the version tags to be 2.1.0 please? We aren't adding any new features to the 2.0 release. Thanks

File: core/src/main/scala/io/delta/tables/DeltaTable.scala [hunk 2]
```diff
@@ -136,6 +136,15 @@ class DeltaTable private[tables](
     executeHistory(deltaLog, tableId = table.getTableIdentifierIfExists)
   }
 
+  /**
+   * Get the details of a Delta table such as the format, name, and size.
+   *
+   * @since 2.0.0
```
[the remark this answers is not in the document]
Comment (hedibejaoui) [reply core/src/main/scala/io/delta/tables/DeltaTable.scala]: Updated the tag to 2.1.0

**Comment by hedibejaoui** [comment]
@allisonport-db The PR is ready for review. Please let me know if any changes are needed.
Thanks!

**Review by allisonport-db — APPROVED** [review]
sorry for the delay! lgtm

**Pull request closed without merging**